### ISSUE TITLE
Windows 10 update today: init phase of Qiqqa starts throwing CLR Exceptions like crazy

### ISSUE THREAD

**Issue by GerHobbelt** [jimmejardine/qiqqa-open-source#94]
```
'Qiqqa.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_32\mscorlib\v4.0_4.0.0.0__b77a5c561934e089\mscorlib.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: DefaultDomain): Loaded 'W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\Qiqqa.exe'. Symbols loaded.
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\WindowsBase\v4.0_4.0.0.0__31bf3856ad364e35\WindowsBase.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Core\v4.0_4.0.0.0__b77a5c561934e089\System.Core.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System\v4.0_4.0.0.0__b77a5c561934e089\System.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_32\PresentationCore\v4.0_4.0.0.0__31bf3856ad364e35\PresentationCore.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xaml\v4.0_4.0.0.0__b77a5c561934e089\System.Xaml.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\Utilities.dll'. Symbols loaded.
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\log4net.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\PrivateAssemblies\Runtime\Microsoft.VisualStudio.Debugger.Runtime.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Configuration\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Configuration.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Xml\v4.0_4.0.0.0__b77a5c561934e089\System.Xml.dll'. 
20191005.171851 [Q] INFO  [Main] [2.098M] Logging initialised.
  Stacktrace:
       at Utilities.Logging..cctor() in W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Utilities\Logging.cs:line 25
       at Utilities.Logging.Info(String msg, Object[] args)
       at Utilities.Files.FileAssociationRegistration.DoFileExtensionRegistration(String extension_including_dot, String prog_id, String shell_menu_prompt) in W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Utilities\Files\FileAssociationRegistration.cs:line 42
       at Qiqqa.Main.FileAssociationRegistration.DoRegistration() in W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\Main\FileAssociationRegistration.cs:line 7
       at Qiqqa.Main.MainEntry.DoPreamble() in W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\Main\MainEntry.cs:line 143
       at Qiqqa.Main.MainEntry..cctor() in W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\Main\MainEntry.cs:line 48
       at Qiqqa.Main.MainEntry.Main()
20191005.171851 [Q] ERROR [Main] [2.138M] Logging init failures (due to premature init/usage?):
20191005.171851 [Q] INFO  [Main] [2.147M] [0.182M] Loaded assembly: log4net, Version=2.0.8.0, Culture=neutral, PublicKeyToken=669e0ddf0bb1aa2a
20191005.171851 [Q] INFO  [Main] [2.147M] [0.190M] Loaded assembly: Microsoft.VisualStudio.Debugger.Runtime, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
20191005.171851 [Q] INFO  [Main] [2.147M] [0.190M] Loaded assembly: System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
20191005.171851 [Q] INFO  [Main] [2.155M] [0.198M] Loaded assembly: System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
20191005.171851 [Q] INFO  [Main] [2.155M] Performing file association for .qiqqa_bundle_manifest.
20191005.171851 [Q] INFO  [Main] [2.155M] Loaded assembly: System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
20191005.171851 [Q] INFO  [Main] [2.163M] Performing file association for .qiqqa_import_omnipatents.
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll'. 
20191005.171851 [Q] INFO  [Main] [2.163M] Performing url protocol association for qiqqa.
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Deployment\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Deployment.dll'. 
20191005.171851 [Q] INFO  [Main] [3.783M] Loaded assembly: System.Deployment, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
Exception thrown: 'System.Deployment.Application.InvalidDeploymentException' in System.Deployment.dll
Application identity is not set.

20191005.172117 [Q] INFO  [Main] [3.807M] COMMON STATISTICS:
Current version:       82
Current build:         82.0.7217.34745
Current path:          W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug
System path:           C:\WINDOWS\system32
Command line:          "W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\Qiqqa.exe" 
OS version:            Microsoft Windows NT 6.2.9200.0
CPU count:             8
Machine name:          DESKTOP-IRCRBHC
CLR version:           4.0.30319.42000
Working set:           33Mb
Application data:      C:\Users\Ger\AppData\Roaming
Temp path:             C:\Users\Ger\AppData\Local\Temp\
CurrentCulture:        en-US
CurrentUICulture:      en-US
CurrentUICulture.Name: en-US
InstalledUICulture:    en-US
InvariantCulture:      
UTC offset:            2
.NET4 client installed True
.NET4 full installed   True
Application is not network deployed.

'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework.Aero2\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.Aero2.dll'. 
20191005.172118 [Q] INFO  [Main] [1.679M] Loaded assembly: PresentationFramework.Aero2, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
20191005.172118 [Q] INFO  [Main] [2.018M] Creating ShutdownableManager
20191005.172118 [Q] INFO  [Main] [2.051M] ShutdownableManager is registering Qiqqa.Common.Configuration.ConfigurationManager
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\Users\Ger\AppData\Local\Temp\VisualStudio.XamlDiagnostics.19408\Microsoft.VisualStudio.DesignTools.WpfTap.dll'. 
20191005.172118 [Q] INFO  [Main] [2.330M] Loaded assembly: Microsoft.VisualStudio.DesignTools.WpfTap, Version=16.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Runtime.Serialization\v4.0_4.0.0.0__b77a5c561934e089\System.Runtime.Serialization.dll'. 
20191005.172118 [Q] INFO  [Main] [2.729M] Loaded assembly: System.Runtime.Serialization, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\SMDiagnostics\v4.0_4.0.0.0__b77a5c561934e089\SMDiagnostics.dll'. 
20191005.172118 [Q] INFO  [Main] [2.729M] Loaded assembly: SMDiagnostics, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.ServiceModel.Internals\v4.0_4.0.0.0__31bf3856ad364e35\System.ServiceModel.Internals.dll'. 
20191005.172118 [Q] INFO  [Main] [2.738M] Loaded assembly: System.ServiceModel.Internals, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\UIAutomationProvider\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationProvider.dll'. 
20191005.172118 [Q] INFO  [Main] [2.934M] Loaded assembly: UIAutomationProvider, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
20191005.172118 [Q] INFO  [Main] [2.942M] Loaded assembly: UIAutomationTypes, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\UIAutomationTypes\v4.0_4.0.0.0__31bf3856ad364e35\UIAutomationTypes.dll'. 
20191005.172118 [Q] INFO  [Main] [3.660M] +UpgradeManager is running upgrades
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Drawing\v4.0_4.0.0.0__b03f5f7f11d50a3a\System.Drawing.dll'. 
20191005.172118 [Q] INFO  [Main] [3.999M] Loaded assembly: System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
20191005.172118 [Q] INFO  [Main] [4.736M] Upgrading from 003 to 004
20191005.172118 [Q] INFO  [Main] [4.932M] Upgrading from 012 to 013
20191005.172118 [Q] INFO  [Main] [4.941M] Local Qiqqa directory exists
20191005.172118 [Q] INFO  [Main] [4.941M] NOT moving Qiqqa directory from Roaming to Local
20191005.172118 [Q] INFO  [Main] [4.941M] Loaded assembly: System.Data.SQLite, Version=1.0.111.0, Culture=neutral, PublicKeyToken=db937bc2d44ff139
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\System.Data.SQLite.dll'. 
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_32\System.Data\v4.0_4.0.0.0__b77a5c561934e089\System.Data.dll'. 
20191005.172118 [Q] INFO  [Main] [4.941M] Loaded assembly: System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_32\System.Transactions\v4.0_4.0.0.0__b77a5c561934e089\System.Transactions.dll'. 
20191005.172118 [Q] INFO  [Main] [4.949M] Loaded assembly: System.Transactions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
20191005.172118 [Q] INFO  [Main] [4.949M] Upgrading from 037 to 038
20191005.172118 [Q] INFO  [Main] [4.957M] Inspecting directory D:\Qiqqa\base\1B9D5176-E568-4BB7-98F1-C09A5DBE5C3D
20191005.172118 [Q] INFO  [Main] [4.957M] Inspecting directory D:\Qiqqa\base\3A614D8C-0882-4CAB-8FBD-A9E494093283
20191005.172118 [Q] INFO  [Main] [4.957M] Inspecting directory D:\Qiqqa\base\5FB85694-536D-4A90-AF13-0306AEE091B6
20191005.172118 [Q] INFO  [Main] [4.957M] Inspecting directory D:\Qiqqa\base\B91A9A6C-07EC-4789-AA77-9953DCB4FD55
20191005.172118 [Q] INFO  [Main] [4.965M] Inspecting directory D:\Qiqqa\base\C9F0D079-EE4C-4A15-8547-72164A7A356D
20191005.172118 [Q] INFO  [Main] [4.965M] Inspecting directory D:\Qiqqa\base\E5283F53-4D1B-40EF-B526-5626BC77F232
20191005.172118 [Q] INFO  [Main] [4.965M] Inspecting directory D:\Qiqqa\base\Guest
20191005.172118 [Q] INFO  [Main] [4.965M] Inspecting directory D:\Qiqqa\base\Guest2
20191005.172118 [Q] INFO  [Main] [4.965M] Inspecting directory D:\Qiqqa\base\INTRANET-EVIL-PDF-SAMPLES-FOR-QIQQA
20191005.172118 [Q] INFO  [Main] [4.973M] Inspecting directory D:\Qiqqa\base\INTRANET-OLD-BIG-DB
20191005.172118 [Q] INFO  [Main] [4.973M] Inspecting directory D:\Qiqqa\base\INTRANET_0D4E8DC2-412B-461D-B359-63E40F1359B9
20191005.172118 [Q] INFO  [Main] [4.973M] Inspecting directory D:\Qiqqa\base\INTRANET_C193EB5E-1C94-40F1-BCF9-4FF596ECA5F9
20191005.172118 [Q] INFO  [Main] [4.973M] Inspecting directory D:\Qiqqa\base\INTRANET_E57B9774-4712-430E-93E0-E67433F7DF07
20191005.172118 [Q] INFO  [Main] [4.973M] Inspecting directory D:\Qiqqa\base\INTRANET_EF52564A-831D-42F2-B956-815CF0418C08
20191005.172118 [Q] INFO  [Main] [4.982M] Inspecting directory D:\Qiqqa\base\ocr
20191005.172118 [Q] INFO  [Main] [4.982M] Inspecting directory D:\Qiqqa\base\UnitTest
20191005.172118 [Q] INFO  [Main] [4.982M] Inspecting directory D:\Qiqqa\base\xulrunner-33
20191005.172118 [Q] INFO  [Main] [5.039M] -UpgradeManager is running upgrades
20191005.172119 [Q] INFO  [Main] [5.416M] +static Main()
'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'W:\Users\Ger\Projects\sites\library.visyond.gov\80\lib\tooling\qiqqa\Qiqqa\bin\Debug\icons.dll'. Symbols loaded.
20191005.172119 [Q] INFO  [Main] [5.432M] Loaded assembly: icons, Version=82.0.0.0, Culture=neutral, PublicKeyToken=null
Exception thrown: 'System.Deployment.Application.InvalidDeploymentException' in System.Deployment.dll
Application identity is not set.

'Qiqqa.exe' (CLR v4.0.30319: Qiqqa.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework-SystemXml\v4.0_4.0.0.0__b77a5c561934e089\PresentationFramework-SystemXml.dll'. 
20191005.172146 [Q] INFO  [Main] [5.927M] Loaded assembly: PresentationFramework-SystemXml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089
Exception thrown: 'System.FormatException' in PresentationCore.dll
Token is not valid.

The thread 0x5248 has exited with code 0 (0x0).
The thread 0x58c8 has exited with code 0 (0x0).
Exception thrown: 'System.InvalidCastException' in PresentationFramework.dll
Unable to cast object of type 'System.Collections.Hashtable' to type 'System.String'.

An exception of type 'System.InvalidCastException' occurred in PresentationFramework.dll and wasn't handled before a managed/native boundary
Unable to cast object of type 'System.Collections.Hashtable' to type 'System.String'.

Exception thrown: 'System.Reflection.TargetInvocationException' in mscorlib.dll
Exception has been thrown by the target of an invocation.

Exception thrown: 'System.FormatException' in PresentationCore.dll
Token is not valid.

Exception thrown: 'System.FormatException' in PresentationCore.dll
Token is not valid.
```

**Comment by GerHobbelt** [comment]
(Note: those exceptions occur in the libraries used by Qiqqa and are shown by the MSVS debugger, but these do not crash Qiqqa. It's just that the debugger is suddenly very loud while it was rather silent before.)

**Comment by GerHobbelt** [comment]
No change. Still a nuisance and no way to fix it with MSVS 2019 v16.0.3 release. Case closed. 😞
